pinot-core/src/main/java/org/apache/pinot/core/util/GenericRowSerDeUtils.java

```diff
@@ -0,0 +1,316 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.util;
+
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import org.apache.pinot.segment.local.segment.memory.PinotDataBuffer;
+import org.apache.pinot.spi.data.FieldSpec;
+import org.apache.pinot.spi.data.readers.GenericRow;
+
+
+/**
+ * Utility methods for serde of {@link GenericRow}
+ * Deserialization assumes it is deserializing from a {@link PinotDataBuffer}
+ */
+public final class GenericRowSerDeUtils {
+
+  private GenericRowSerDeUtils() {
+
+  }
+
+  /**
+   * Serialize the given GenericRow. The data is stored in native byte order.
+   * @param genericRow GenericRow to serialize
+   * @param fieldSpecs the fields to serialize
+   * @return serialized bytes
+   */
+  public static byte[] serializeGenericRow(GenericRow genericRow, List<FieldSpec> fieldSpecs) {
+    int numBytes = 0;
+
+    for (FieldSpec fieldSpec : fieldSpecs) {
+      Object value = genericRow.getValue(fieldSpec.getName());
+
+      if (fieldSpec.isSingleValueField()) {
+        switch (fieldSpec.getDataType().getStoredType()) {
+
+          case INT:
+            numBytes += Integer.BYTES;
+            break;
+          case LONG:
+            numBytes += Long.BYTES;
+            break;
+          case FLOAT:
+            numBytes += Float.BYTES;
+            break;
+          case DOUBLE:
+            numBytes += Double.BYTES;
+            break;
+          case STRING:
+            byte[] stringBytes = ((String) value).getBytes(StandardCharsets.UTF_8);
```

Comment:
Use `StringUtils.encodeUtf8()` which is faster per #3073

Reply:
oh nice, thanks for the tip